src/compiler/checker.ts

```diff
@@ -6569,11 +6569,12 @@ namespace ts {
             return result;
         }
 
-        function isJSDocOptionalParameter(node: ParameterDeclaration) {
+        function isJSDocOptionalParameter(node: ParameterDeclaration, skipType?: boolean) {
             return isInJavaScriptFile(node) && (
                 node.type && node.type.kind === SyntaxKind.JSDocOptionalType

```

Comment:
Might want to comment that this case is only for the parameters of a `JSDocFunctionType`, otherwise `.type` should always be TS syntax
Also, why does this one not look for `JSDocNullableType` too?

Reply:
Added. The comment. See @weswigham's report of our discussion for why `skipType` is incorrect &mdash; I'll push an update shortly.